node-data/src/lib.rs

```diff
@@ -17,20 +17,102 @@ pub trait Serializable {
     where
         Self: Sized;
 
+    /// Write varint fields
+    fn write_varint<W: Write>(w: &mut W, v: u64) -> io::Result<()> {
+        if v < 0xfd {
+            w.write_all(&[v as u8])?
+        } else if v <= 1 << (16 - 1) {
+            w.write_all(&[0xfd])?;
+            w.write_all(&(v as u16).to_le_bytes())?;
+        } else if v <= 1 << (32 - 1) {
+            w.write_all(&[0xfe])?;
+            w.write_all(&(v as u32).to_le_bytes())?;
+        } else {
+            w.write_all(&[0xff])?;
+            w.write_all(&v.to_le_bytes())?;
+        }
+        Ok(())
+    }
+
+    /// Write length-prefixed fields
+    fn write_var_bytes<W: Write>(w: &mut W, buf: &[u8]) -> io::Result<()> {
+        Self::write_varint(w, buf.len() as u64)?;
+        w.write_all(buf)
+    }
+
+    /// Reads varint fields
+    fn read_varint<R: Read>(r: &mut R) -> io::Result<usize> {
+        let mut buf = [0u8; 1];
+        r.read_exact(&mut buf)?;
+        let d = buf[0];
+
+        let len = match d {
+            0xff => {
+                let mut buf = [0u8; 8];
+                r.read_exact(&mut buf)?;
+                u64::from_le_bytes(buf) as usize
+            }
+            0xfe => {
+                let mut buf = [0u8; 4];
+                r.read_exact(&mut buf)?;
+                u32::from_le_bytes(buf) as usize
+            }
+            0xfd => {
+                let mut buf = [0u8; 2];
+                r.read_exact(&mut buf)?;
+                u16::from_le_bytes(buf) as usize
+            }
+            val => val as usize,
+        };
+
+        Ok(len)
+    }
+
+    /// Reads length-prefixed fields
+    fn read_var_bytes<R: Read>(r: &mut R) -> io::Result<Vec<u8>> {
+        let len = Self::read_varint(r)?;
+
+        let mut buf = vec![0u8; len];
+        r.read_exact(&mut buf)?;
+
+        Ok(buf)
+    }
+
     /// Writes length-prefixed fields
     fn write_var_le_bytes<W: Write>(w: &mut W, buf: &[u8]) -> io::Result<()> {
```

Comment:
Do we need  `write_var_le_bytes` anymore? I thought it's completely deprecated in favor of `write_var_bytes`?

Reply:
I'm for leaving those there, since the removal should be part of the encoding refactor. Going to tackle it later on